### ISSUE TITLE
ArrayBracketSpacing not working?

### ISSUE THREAD

**Issue by dingo-d** [squizlabs/PHP_CodeSniffer#3117]
**Describe the bug**
I'm building a custom ruleset, and I wanted to include the `Squiz.Arrays.ArrayBracketSpacing` sniff in it. When I test this against the code example, it doesn't report any issues.

**Code sample**
```php
<?php

$arr = [ 'some value', 'some other value' ]; // Error
$arr2 = ['some value', 'some other value']; // Ok.
$arr3 = array( 'some value', 'some other value' ); // Error.
$arr4 = array('some value', 'some other value'); // Ok.
```

**Custom ruleset**
I've even removed all other rules from the standard to check what is happening (full ruleset is [here](https://github.com/infinum/coding-standards-wp/tree/feature/wpcs-update-bc))

```xml
<?xml version="1.0"?>
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="Infinum" namespace="Infinum"
         xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/squizlabs/PHP_CodeSniffer/master/phpcs.xsd">
    <description>Infinum coding standards for WordPress projects</description>

    <config name="encoding" value="utf-8"/>

    <!-- Exclude certain patterns. -->
    <exclude-pattern>/vendor/*</exclude-pattern>
    <exclude-pattern>/node_modules/*</exclude-pattern>

    <!-- Only check php code, linters are used for checking js and css. -->
    <arg name="extensions" value="php"/>

    <!-- Enable colors in report. -->
    <arg name="colors"/>

    <!-- Add source codes in the report. -->
    <arg value="sp"/>

    <!-- Disallow spacing inside arrays. -->
    <rule ref="Squiz.Arrays.ArrayBracketSpacing"/>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `./vendor/squizlabs/php_codesniffer/bin/phpcs --standard=./Infinum ./test.php -vv`
3. See results

<details>
<sumary>PHPCS output</summary>
<pre><code>
Processing ruleset /Users/denis.zoljom/Projects/Infinum/coding-standards-wp/Infinum/ruleset.xml
	Adding sniff files from /Users/denis.zoljom/Projects/Infinum/coding-standards-wp/Infinum/Sniffs directory
		=> /Users/denis.zoljom/Projects/Infinum/coding-standards-wp/Infinum/Sniffs/Shortcodes/DisallowDoShortcodeSniff.php
	=> set config value encoding: utf-8
	Processing rule "Squiz.Arrays.ArrayBracketSpacing"
		=> /Users/denis.zoljom/Projects/Infinum/coding-standards-wp/vendor/squizlabs/php_codesniffer/src/Standards/Squiz/Sniffs/Arrays/ArrayBracketSpacingSniff.php
	=> set command line value --extensions=php
	=> set command line value --colors
	=> set command line value -sp
	=> added global absolute ignore pattern: /vendor/*
	=> added global absolute ignore pattern: /node_modules/*
=> Ruleset processing complete; included 2 sniffs and excluded 0
Creating file list... DONE (1 files in queue)
Changing into directory /Users/denis.zoljom/Projects/Infinum/coding-standards-wp/tests
Processing code-example-5.php
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php\n
	Process token [1]: T_WHITESPACE => \n
	Process token [2]: T_DOC_COMMENT => /**\n·*·Test\n·*\n·*·@package·Sniffs\n·*/
		*** START COMMENT TOKENIZING ***
		Create comment token: T_DOC_COMMENT_OPEN_TAG => /**
		Create comment token: T_DOC_COMMENT_WHITESPACE => \n
		Create comment token: T_DOC_COMMENT_WHITESPACE => ·
		Create comment token: T_DOC_COMMENT_STAR => *
		Create comment token: T_DOC_COMMENT_WHITESPACE => ·
		Create comment token: T_DOC_COMMENT_STRING => Test
		Create comment token: T_DOC_COMMENT_WHITESPACE => \n
		Create comment token: T_DOC_COMMENT_WHITESPACE => ·
		Create comment token: T_DOC_COMMENT_STAR => *
		Create comment token: T_DOC_COMMENT_WHITESPACE => \n
		Create comment token: T_DOC_COMMENT_WHITESPACE => ·
		Create comment token: T_DOC_COMMENT_STAR => *
		Create comment token: T_DOC_COMMENT_WHITESPACE => ·
		Create comment token: T_DOC_COMMENT_TAG => @package
		Create comment token: T_DOC_COMMENT_WHITESPACE => ·
		Create comment token: T_DOC_COMMENT_STRING => Sniffs
		Create comment token: T_DOC_COMMENT_WHITESPACE => \n
		Create comment token: T_DOC_COMMENT_WHITESPACE => ·
		Create comment token: T_DOC_COMMENT_CLOSE_TAG => */
		*** END COMMENT TOKENIZING ***
	Process token [3]: T_WHITESPACE => \n\n
	Process token [4]: T_VARIABLE => $arr
	Process token [5]: T_WHITESPACE => ·
	Process token  6 : T_EQUAL => =
	Process token [7]: T_WHITESPACE => ·
	Process token  8 : T_OPEN_SQUARE_BRACKET => [
	Process token [9]: T_WHITESPACE => ·
	Process token [10]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token  11 : T_COMMA => ,
	Process token [12]: T_WHITESPACE => ·
	Process token [13]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token [14]: T_WHITESPACE => ·
	Process token  15 : T_CLOSE_SQUARE_BRACKET => ]
	Process token  16 : T_SEMICOLON => ;
	Process token [17]: T_WHITESPACE => \n
	Process token [18]: T_VARIABLE => $arr2
	Process token [19]: T_WHITESPACE => ·
	Process token  20 : T_EQUAL => =
	Process token [21]: T_WHITESPACE => ·
	Process token  22 : T_OPEN_SQUARE_BRACKET => [
	Process token [23]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token  24 : T_COMMA => ,
	Process token [25]: T_WHITESPACE => ·
	Process token [26]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token  27 : T_CLOSE_SQUARE_BRACKET => ]
	Process token  28 : T_SEMICOLON => ;
	Process token [29]: T_WHITESPACE => \n
	Process token [30]: T_VARIABLE => $arr3
	Process token [31]: T_WHITESPACE => ·
	Process token  32 : T_EQUAL => =
	Process token [33]: T_WHITESPACE => ·
	Process token [34]: T_ARRAY => array
	Process token  35 : T_OPEN_PARENTHESIS => (
	Process token [36]: T_WHITESPACE => ·
	Process token [37]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token  38 : T_COMMA => ,
	Process token [39]: T_WHITESPACE => ·
	Process token [40]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token [41]: T_WHITESPACE => ·
	Process token  42 : T_CLOSE_PARENTHESIS => )
	Process token  43 : T_SEMICOLON => ;
	Process token [44]: T_WHITESPACE => \n
	Process token [45]: T_VARIABLE => $arr4
	Process token [46]: T_WHITESPACE => ·
	Process token  47 : T_EQUAL => =
	Process token [48]: T_WHITESPACE => ·
	Process token [49]: T_ARRAY => array
	Process token  50 : T_OPEN_PARENTHESIS => (
	Process token [51]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token  52 : T_COMMA => ,
	Process token [53]: T_WHITESPACE => ·
	Process token [54]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token  55 : T_CLOSE_PARENTHESIS => )
	Process token  56 : T_SEMICOLON => ;
	Process token [57]: T_WHITESPACE => \n
	*** END PHP TOKENIZING ***
	*** START TOKEN MAP ***
	=> Found square bracket opener at 27
	=> Found square bracket closer at 34 for 27
	=> Found square bracket opener at 41
	=> Found square bracket closer at 46 for 41
	=> Found parenthesis owner at 53
	=> Found parenthesis opener at 54 for 53
	=> Found parenthesis closer at 61 for 53
	=> Found parenthesis owner at 68
	=> Found parenthesis opener at 69 for 68
	=> Found parenthesis closer at 74 for 68
	*** END TOKEN MAP ***
	*** START SCOPE MAP ***
	*** END SCOPE MAP ***
	*** START LEVEL MAP ***
	Process token 0 on line 1 [col:1;len:5;lvl:0;]: T_OPEN_TAG => <?php\n
	Process token 1 on line 2 [col:1;len:0;lvl:0;]: T_WHITESPACE => \n
	Process token 2 on line 3 [col:1;len:3;lvl:0;]: T_DOC_COMMENT_OPEN_TAG => /**
	Process token 3 on line 3 [col:4;len:0;lvl:0;]: T_DOC_COMMENT_WHITESPACE => \n
	Process token 4 on line 4 [col:1;len:1;lvl:0;]: T_DOC_COMMENT_WHITESPACE => ·
	Process token 5 on line 4 [col:2;len:1;lvl:0;]: T_DOC_COMMENT_STAR => *
	Process token 6 on line 4 [col:3;len:1;lvl:0;]: T_DOC_COMMENT_WHITESPACE => ·
	Process token 7 on line 4 [col:4;len:4;lvl:0;]: T_DOC_COMMENT_STRING => Test
	Process token 8 on line 4 [col:8;len:0;lvl:0;]: T_DOC_COMMENT_WHITESPACE => \n
	Process token 9 on line 5 [col:1;len:1;lvl:0;]: T_DOC_COMMENT_WHITESPACE => ·
	Process token 10 on line 5 [col:2;len:1;lvl:0;]: T_DOC_COMMENT_STAR => *
	Process token 11 on line 5 [col:3;len:0;lvl:0;]: T_DOC_COMMENT_WHITESPACE => \n
	Process token 12 on line 6 [col:1;len:1;lvl:0;]: T_DOC_COMMENT_WHITESPACE => ·
	Process token 13 on line 6 [col:2;len:1;lvl:0;]: T_DOC_COMMENT_STAR => *
	Process token 14 on line 6 [col:3;len:1;lvl:0;]: T_DOC_COMMENT_WHITESPACE => ·
	Process token 15 on line 6 [col:4;len:8;lvl:0;]: T_DOC_COMMENT_TAG => @package
	Process token 16 on line 6 [col:12;len:1;lvl:0;]: T_DOC_COMMENT_WHITESPACE => ·
	Process token 17 on line 6 [col:13;len:6;lvl:0;]: T_DOC_COMMENT_STRING => Sniffs
	Process token 18 on line 6 [col:19;len:0;lvl:0;]: T_DOC_COMMENT_WHITESPACE => \n
	Process token 19 on line 7 [col:1;len:1;lvl:0;]: T_DOC_COMMENT_WHITESPACE => ·
	Process token 20 on line 7 [col:2;len:2;lvl:0;]: T_DOC_COMMENT_CLOSE_TAG => */
	Process token 21 on line 7 [col:4;len:0;lvl:0;]: T_WHITESPACE => \n
	Process token 22 on line 8 [col:1;len:0;lvl:0;]: T_WHITESPACE => \n
	Process token 23 on line 9 [col:1;len:4;lvl:0;]: T_VARIABLE => $arr
	Process token 24 on line 9 [col:5;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 25 on line 9 [col:6;len:1;lvl:0;]: T_EQUAL => =
	Process token 26 on line 9 [col:7;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 27 on line 9 [col:8;len:1;lvl:0;]: T_OPEN_SQUARE_BRACKET => [
	Process token 28 on line 9 [col:9;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 29 on line 9 [col:10;len:12;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token 30 on line 9 [col:22;len:1;lvl:0;]: T_COMMA => ,
	Process token 31 on line 9 [col:23;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 32 on line 9 [col:24;len:18;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token 33 on line 9 [col:42;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 34 on line 9 [col:43;len:1;lvl:0;]: T_CLOSE_SQUARE_BRACKET => ]
	Process token 35 on line 9 [col:44;len:1;lvl:0;]: T_SEMICOLON => ;
	Process token 36 on line 9 [col:45;len:0;lvl:0;]: T_WHITESPACE => \n
	Process token 37 on line 10 [col:1;len:5;lvl:0;]: T_VARIABLE => $arr2
	Process token 38 on line 10 [col:6;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 39 on line 10 [col:7;len:1;lvl:0;]: T_EQUAL => =
	Process token 40 on line 10 [col:8;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 41 on line 10 [col:9;len:1;lvl:0;]: T_OPEN_SQUARE_BRACKET => [
	Process token 42 on line 10 [col:10;len:12;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token 43 on line 10 [col:22;len:1;lvl:0;]: T_COMMA => ,
	Process token 44 on line 10 [col:23;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 45 on line 10 [col:24;len:18;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token 46 on line 10 [col:42;len:1;lvl:0;]: T_CLOSE_SQUARE_BRACKET => ]
	Process token 47 on line 10 [col:43;len:1;lvl:0;]: T_SEMICOLON => ;
	Process token 48 on line 10 [col:44;len:0;lvl:0;]: T_WHITESPACE => \n
	Process token 49 on line 11 [col:1;len:5;lvl:0;]: T_VARIABLE => $arr3
	Process token 50 on line 11 [col:6;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 51 on line 11 [col:7;len:1;lvl:0;]: T_EQUAL => =
	Process token 52 on line 11 [col:8;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 53 on line 11 [col:9;len:5;lvl:0;]: T_ARRAY => array
	Process token 54 on line 11 [col:14;len:1;lvl:0;]: T_OPEN_PARENTHESIS => (
	Process token 55 on line 11 [col:15;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 56 on line 11 [col:16;len:12;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token 57 on line 11 [col:28;len:1;lvl:0;]: T_COMMA => ,
	Process token 58 on line 11 [col:29;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 59 on line 11 [col:30;len:18;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token 60 on line 11 [col:48;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 61 on line 11 [col:49;len:1;lvl:0;]: T_CLOSE_PARENTHESIS => )
	Process token 62 on line 11 [col:50;len:1;lvl:0;]: T_SEMICOLON => ;
	Process token 63 on line 11 [col:51;len:0;lvl:0;]: T_WHITESPACE => \n
	Process token 64 on line 12 [col:1;len:5;lvl:0;]: T_VARIABLE => $arr4
	Process token 65 on line 12 [col:6;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 66 on line 12 [col:7;len:1;lvl:0;]: T_EQUAL => =
	Process token 67 on line 12 [col:8;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 68 on line 12 [col:9;len:5;lvl:0;]: T_ARRAY => array
	Process token 69 on line 12 [col:14;len:1;lvl:0;]: T_OPEN_PARENTHESIS => (
	Process token 70 on line 12 [col:15;len:12;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·value'
	Process token 71 on line 12 [col:27;len:1;lvl:0;]: T_COMMA => ,
	Process token 72 on line 12 [col:28;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 73 on line 12 [col:29;len:18;lvl:0;]: T_CONSTANT_ENCAPSED_STRING => 'some·other·value'
	Process token 74 on line 12 [col:47;len:1;lvl:0;]: T_CLOSE_PARENTHESIS => )
	Process token 75 on line 12 [col:48;len:1;lvl:0;]: T_SEMICOLON => ;
	Process token 76 on line 12 [col:49;len:0;lvl:0;]: T_WHITESPACE => \n
	*** END LEVEL MAP ***
	*** START ADDITIONAL PHP PROCESSING ***
	* token 41 on line 10 changed from T_OPEN_SQUARE_BRACKET to T_OPEN_SHORT_ARRAY
	* token 46 on line 10 changed from T_CLOSE_SQUARE_BRACKET to T_CLOSE_SHORT_ARRAY
	* token 27 on line 9 changed from T_OPEN_SQUARE_BRACKET to T_OPEN_SHORT_ARRAY
	* token 34 on line 9 changed from T_CLOSE_SQUARE_BRACKET to T_CLOSE_SHORT_ARRAY
	*** END ADDITIONAL PHP PROCESSING ***
[PHP => 77 tokens in 12 lines]...
DONE in 10ms (0 errors, 0 warnings)
Time: 42ms; Memory: 4MB
</code></pre>
</details>

**Expected behavior**
I should get 2 errors because the example has spacing when opening the array brackets.

**Versions (please complete the following information):**
 - OS: MacOS 10-15.6
 - PHP: 7.4.8
 - PHPCS: 3.5.6
 - Standard: Squiz plus one custom made sniff.

**Comment by dingo-d** [comment]
Ok, this is weird. I added an error log in the sniff `register()` method, and I get the output, but nothing from `process()` method. Could there be some issues with the tokens?

**Comment by dingo-d** [comment]
I added error log inside the `Runner.php` on line 630 `error_log(print_r($file, true));`, and when examinig that object, the part with the above sniff is empty.
```
[PHP_CodeSniffer\Standards\Squiz\Sniffs\Arrays\ArrayBracketSpacingSniff] => PHP_CodeSniffer\Standards\Squiz\Sniffs\Arrays\ArrayBracketSpacingSniff Object
                        (
                        )
```

Still no idea why 😅

**Comment by jrfnl** [comment]
@dingo-d The `Squiz.Arrays.ArrayBracketSpacing` sniff is not about he spacing around the brackets of an array declaration, but about the spacing around the square brackekts used for _array access_, i.e. `$array['access'].

Within PHPCS itself, you'd end up with the `Squiz.Array.ArrayDeclaration` if you want to array declaration brackets checked, but that sniff does way more and is quite buggy.

You may to give the (configurable) [`NormalizedArrays.Arrays.ArrayBraceSpacing`](https://github.com/PHPCSStandards/PHPCSExtra#normalizedarraysarraysarraybracespacing-wrench-bar_chart-books) from PHPCSExtra a try.

**Comment by dingo-d** [comment]
> @dingo-d The Squiz.Arrays.ArrayBracketSpacing sniff is not about the spacing around the brackets of an array declaration, but about the spacing around the square brackekts used for array access, i.e. $array['access'].

Oh, I totally missed the `$` in the documentation. I thought it was about the array spacing within the brackets (the naming threw me off).

I'll try with the extra 👍
